### DISCUSSION TITLE
Ubuntu merge (sort of) complete - 6.12 release

### DISCUSSION THREAD

**Discussion by jglathe** [jglathe/linux_ms_dev_kit#29] [Announcements]
An update on the state of things. We're sort of reaching maturity:

- we're on kernel 6.12 now, on the [qcom-x1e-6.12](https://github.com/jglathe/linux_ms_dev_kit/tree/jg/ubuntu-qcom-x1e-6.12) branch of Ubuntu Concept X1E 24.10. The normal Ubuntu config works with all of them. HP Omnibook is an install target that will be recognized by the installer ISO.  
- Windows Dev Kit 2023 device tree is submitted for upstreaming.
- HP Omnibook X14 device tree is submitted for upstreaming.
- Board files for both are submitted for upstreaming.

With Canonical picking up the ball it will be sort of easier to get Linux installed on all of these boxes. It may also mean focussing on the things that are still an issue / don't work yet:

- Audio (built-in devices and via DisplayPort)
- DP Altmode
- RTC on the X14
- keyboard backlight and function keys on the X14

I also have a SnapDragon Dev Kit sitting here, with really unwilling DP Altmode, so no display yet. Still enough to do.

**Comment by Shinmegumi** [comment]
So, for example, is the new 25.04 release able to just be installed on the WDK 2023 or do we still need to use one of these builds and just update the kernel?

Or are there new builds coming soon of the latest images/kernel?

My Linux knowledge is sadly lacking as I haven't used it much.

**Reply by jglathe** [reply]
Actually, I will verify if its booting from the ISO. Since the device tree is upstream, it should just work :tm: Although on my kernel packages it has a few more small tweaks, like dp altmode link frequencies to support HBR3.

I am (still) constantly updating my tree (in this repo) and occasionally publish pre-built kernel packages with Ubuntu build config.

**Reply by jglathe** [reply]
Okay I just tried. Ubuntu 25.04 ISO doesn't recognize the Windows Dev Kit 2023, gives a warning, doesn't boot. That's sort of a pity. I have a [bug on launchpad](https://bugs.launchpad.net/ubuntu-concept/+bug/2084975) open for this, will bump it. The device tree has been upstreamed a while ago...

**Reply by Shinmegumi** [reply]
I appreciate it. Are you doing any new builds or LTS builds with the new kernel for the WDK 2023? I was looking to load Linux back up on it again.

**Reply by jglathe** [reply]
The [old 24.10 image](https://drive.google.com/file/d/13rkHGODaK94HHgCIB8_zdbCrgnDkwwNs/view?usp=sharing) is booting nicely, upgrading to 25.04 with `do-release-upgrade -d` (currently) is no problem, works fine. I try to get off these images, they are a bit time consuming to build.

**Reply by Shinmegumi** [reply]
Do you think I can upgrade the kernel to 6.14 or will it break things?

**Reply by jglathe** [reply]
You should be able to upgade to 6.15.0, too. Using it on the wdk23 with 25.04. It has good patches for ath11k, and lots of DP altmode improvoment. Venus will now be officially mainlined, so this might give some extra fun.

**Reply by Shinmegumi** [reply]
Awesome. This may seem like a silly question, but do I need to disable Bitlocker since it is activated by default on the WDK? I feel like I read somewhere that it pulls information and firmware from the Windows install.

**Reply by jglathe** [reply]
Yes and yes. The boot image needs the unencrypted Windows partition, also secure boot needs to be disabled. I would recommend to update the firmware on Windows, too. Microsoft is still releasing updates and they at least don't hurt.

**Reply by Shinmegumi** [reply]
I found the guide you did. I saw that it runs from the USB. Is it possible to install it on the SSD alongside Windows? I shrunk my Windows install as I don't intend to use it.

**Reply by Shinmegumi** [reply]
I think I was able to do this. Between conversations you've had with others on here and Google I was able to copy over my partition from the external to the internal SSD and then update the boot after installing grub to the SSD.

**Reply by jglathe** [reply]
Yes, I use it with dual boot on the internal SSD for 2 years now.I can only recommend to make a backup of essential data first, otherwise its sort of straight forward. I guess I have described this several times in this repo, it involves creating the partition on the SSD, adding the UEFI boot entry, moving over the boot to the SSD EFI partition, righting the grub installation. Regardless of actual machine, its the same.

**Reply by Shinmegumi** [reply]
> Yes, I use it with dual boot on the internal SSD for 2 years now.I can only recommend to make a backup of essential data first, otherwise its sort of straight forward. I guess I have described this several times in this repo, it involves creating the partition on the SSD, adding the UEFI boot entry, moving over the boot to the SSD EFI partition, righting the grub installation. Regardless of actual machine, its the same.

It was easier than I expected, but was daunting for someone who had no idea what they were doing. I think I'm good as long as my /boot/efi mounts properly after reboot.
